webserver/pkgpkr/webservice/views.py

```diff
@@ -255,31 +303,31 @@ def recommendations_service_api(request):
         try:
             json_data = json.loads(request.body)  # request.raw_post_data w/ Django < 1.4
         except json.JSONDecodeError:
-            return HttpResponseServerError("Could not parse JSON.")
+            return HttpResponseBadRequest("Could not parse JSON.")
 
         # Fetch non-optional keys
         try:
             dependencies = json_data['dependencies']
             language = json_data['language'].lower()
         except KeyError:
-            return HttpResponseServerError('Required JSON keys: `dependencies`, `language`')
+            return HttpResponseBadRequest('Required JSON keys: `dependencies`, `language`')
         except AttributeError as e:
-            return HttpResponseServerError(f'Error casting language to lower(): {e}')
+            return HttpResponseBadRequest(f'Error casting language to lower(): {e}')
 
         # Assure proper inputs
-        if language == PYTHON and (not isinstance(dependencies, list) or not dependencies):
-            return HttpResponseServerError('Python dependencies must be non-empty and of type LIST (i.e. [...]).')
-
-        if language == JAVASCRIPT and (not isinstance(dependencies, dict) or not dependencies):
-            return HttpResponseServerError('Javascript dependencies must be non-empty and of type DICT (i.e. {...}).')
+        if not isinstance(dependencies, list) or not dependencies:
+            return HttpResponseBadRequest(f'{language.capitalize()} dependencies must be non-empty and of type LIST (i.e. [...]).')
 
         # Convert comma separated dependencies into proper expected format
         if language == PYTHON:
             dependencies = '\n'.join(dependencies)
         elif language == JAVASCRIPT:
-            dependencies = ','.join([f'"{k}":"{v}"' for k, v in dependencies.items()])
+            # Converts e.g ["lodash:4.17.15","react:16.13.1"] -> '"lodash":"4.17.15","react":"16.13.1"'
+            formatted_dependencies_list = ['"' + dep.replace(":", '":"') + '"' for dep in dependencies]
```

Comment:
Suggest using `@` as the name/version delimiter, to align with PURL format.

```suggestion
            # Converts e.g ["lodash@4.17.15","react@16.13.1"] -> '"lodash":"4.17.15","react":"16.13.1"'
            formatted_dependencies_list = ['"' + dep.replace("@", '":"') + '"' for dep in dependencies]
```